### DISCUSSION TITLE
Heatmap vs GridLayer OnClick?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#7836] [Q&A] [answered]
Hi All!

In looking at trying to get data back out from the [HeatmapLayer](https://deck.gl/docs/api-reference/aggregation-layers/heatmap-layer) via the `onClick `function I believe I have ran across an issue or some inconsistencies... Or maybe this is by design.

The `onClick` function from the [HeatmapLayer](https://deck.gl/docs/api-reference/aggregation-layers/heatmap-layer) does not provide an `object` property back via the callback. Additionally, it provides back a `coordinate` property. This is contrary to the behavior of the GridLayer [codepen](https://codepen.io/jfuehner/pen/xxyZvWp) and/or other _Grid_ variants.

I have put together a HeatmapLayer [codepen](https://codepen.io/jfuehner/pen/VwEeoMB) where I am trying to get the aggregated count back similar to this GridLayer [codepen](https://codepen.io/jfuehner/pen/xxyZvWp).

Perhaps I am missing something? Is this by design the HeatmapLayer does not provide back a `count ` or a `object`? Additionally, does the  `index` property available correlate to a `data` element _(it doesn't appear to)_?

**Comment by Pessimistress** [comment, marked as answer]
HeatmapLayer renders a continous texture. There are no discrete objects to be picked.
